### PR TITLE
Authorize uuid for update_positions on ResourceController

### PR DESCRIPTION
**Pull request by julienanne**
## Summary

In the backend section, from Ref https://github.com/solidusio/solidus/commit/248b39f9b5c797fd0836277c2fdb7fe50380f045 we can't sort object list when we use uuid for the object's id.
We need to allow records save for the case of uuid.
This change need to be backported until the version 3.1.0 if I'm right ;)

Hope it can help.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

**Comment by julienanne** [comment]
Sorry seems the CI have not the function gen_random_uuid() on postgres for instance so this PR can be closed if uuid is not in your vision at short terms ;)

Have a nice day ;)

**Comment by kennyadsl** [comment]
It seems like Postgres needs to load `pgcrypto` extension for `gen_random_uuid` to work. I don't know if we can give for granted that people have that installed, and if we add it, it's a breaking change for people already using Solidus. 

What if we keep the change without the UUIDs test you added? Do you see any cons for those who are not using UUIDs?

**Comment by julienanne** [comment]
@kennyadsl thanks for answer true fact for the breaking change.

Indeed good idea to keep the code even if the spec don't show the need of change but in this particular case it is ok in my opinion.

And I see no cons due to the fact that "already in place" specs are green on my computer. So ok for me for people who are not using UUIDs.

I will modify this PR.

Thanks

**Comment by julienanne** [comment]
@kennyadsl it seems that I have doing something wrong (CI is red) but I don't know what, can you help me please ?

**Comment by kennyadsl** [comment]
Should be fixed when we merge https://github.com/solidusio/solidus/pull/5010, I will ask you for a rebase when done, thanks!

**Comment by kennyadsl** [comment]
@julienanne please rebase now!

**Comment by julienanne** [comment]
@kennyadsl seems better thanks ;)

**Comment by kennyadsl** [comment]
@solidusio/core-team thoughts?

**Review by adammathys — APPROVED** [review]

**Review by spaghetticode — APPROVED** [review]

File: backend/app/controllers/spree/admin/resource_controller.rb [hunk 1]
```diff
@@ -82,7 +82,7 @@ def update_positions
       records = model_class.where(id: positions.keys).to_a
 
       positions.each do |id, index|
-        records.find { |r| r.id == id.to_i }&.set_list_position(index)
+        records.find { |r| r.id.to_s == id }&.set_list_position(index)
```
Comment (waiting-for-dev) [inline backend/app/controllers/spree/admin/resource_controller.rb:85]: Non-blocking, but given we're not testing it, adding a code comment here would be good.

Comment (julienanne) [reply backend/app/controllers/spree/admin/resource_controller.rb:87]: I write a com, good idea indeed when two implementations make the spec green and we want one implementation more than the other. 
But my English is not accurate so feel free to suggest something more accurate if needed ;)
Thanks ;)

**Comment by julienanne** [comment]
No spec red on my local machine, flaky test ?

**Comment by kennyadsl** [comment]
@julienanne thanks for your contribution! 🎉

**Comment by julienanne** [comment]
@kennyadsl Do you need backports for v3.3, v3.2 and v3.1 or you think it is not necessary ?
If yes if I remember right I have to do one PR per branch.

Have a nice day

**Comment by kennyadsl** [comment]
The backport would have been automatic if I remembered to add the proper labels to the PR, I'm sorry! Let me see if we can trigger the workflow anyway after the PR is merged.

**Comment by kennyadsl** [comment]
@waiting-for-dev suggested to add the labels even if the PR is merged, and it should work. Let me try!

**Comment by kennyadsl** [comment]
Backport done @julienanne! 3.1 is not supported anymore following our policy [here](https://solidus.io/security/), though.

**Comment by julienanne** [comment]
Thanks 
Have a nice day

**Pull request merged**